File: src/Services/Sales/Sales.Domain/ValueObjects/OrderStatus.cs
namespace Sales.Domain;

/// <summary>
/// lifecycle status of an <see cref="Order"/>.
/// </summary>
public enum OrderStatus
{
    /// <summary>The order has been created but confirmation has not been requested yet.</summary>
    Draft,

    /// <summary>Confirmation has been requested and Inventory has not yet responded.</summary>
    PendingInventory,

    /// <summary>Inventory reserved stock for every line and the order is confirmed.</summary>
    Confirmed,

    /// <summary>Inventory rejected the reservation request, typically due to insufficient stock.</summary>
    InventoryRejected,

    /// <summary>The order was cancelled.</summary>
    Cancelled
}

File: src/Web/Sales.Web/src/app/features/orders/constants/order-status.ts
import { StatusDisplay, toStatusDisplay } from '../../../shared/models/status-display.model';

/** String union, not a TS enum: values are backend codes and must match exactly. */
export type OrderStatus = 'Draft' | 'PendingInventory' | 'Confirmed' | 'Cancelled' | 'InventoryRejected';

export const orderStatusDisplays: Readonly<Record<OrderStatus, StatusDisplay>> = {
  Draft: { label: 'Draft', tone: 'neutral' },
  PendingInventory: { label: 'Pending inventory', tone: 'info' },
  Confirmed: { label: 'Confirmed', tone: 'success' },
  Cancelled: { label: 'Cancelled', tone: 'danger' },
  InventoryRejected: { label: 'Inventory rejected', tone: 'danger' }
};

export function orderStatusDisplay(status: string | null | undefined): StatusDisplay {
  return toStatusDisplay(status, orderStatusDisplays);
}

/** User-facing explanation of a realtime status transition. */
export function describeOrderStatusChange(status: string): string {
  switch (status) {
    case 'Confirmed':
      return 'The order was confirmed and inventory was reserved.';
    case 'InventoryRejected':
      return 'The order could not be confirmed because Inventory rejected the reservation.';
    case 'Cancelled':
      return 'The order was cancelled.';
    default:
      return 'The order has new data.';
  }
}
